### ISSUE TITLE
DataVolume importer pod is in pending state

### ISSUE THREAD

**Issue by ksimon1** [kubevirt/containerized-data-importer#1418]
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug
**What happened**:
When I try to create this dataVolume
```
oc apply -f - <<EOF
apiVersion: cdi.kubevirt.io/v1beta1
kind: DataVolume
metadata:
  name: fedora-datavolume-original
spec:
  source:
    registry:
      url: "docker://quay.io/kubevirt/common-templates:fedora"
  pvc:
    accessModes:
      - ReadWriteMany
    resources:
      requests:
        storage: 15Gi
EOF
```
the newly created pod stays in pending state.
```
[ksimon@localhost common-templates]$ oc get pods
NAME                                       READY   STATUS    RESTARTS   AGE
importer-fedora-datavolume-original        0/1     Pending   0          3m17s
```
When I switch `accessModes` to `ReadWriteOnce`, the pod starts without any problem.
**What you expected to happen**:
The dataVolume importer pod imports containerDisk without error

**Environment**:
- CDI version (use `kubectl get deployments cdi-deployment -o yaml`):
v1.24.0
- Kubernetes version (use `kubectl version`):
```
[ksimon@localhost common-templates]$ oc version
Client Version: 4.5.7
Server Version: 4.6.0-fc.9
Kubernetes Version: v1.19.0+1fc699e
```
```
[ksimon@localhost common-templates]$ oc describe pod importer-fedora-datavolume-original
Name:           importer-fedora-datavolume-original
Namespace:      kubevirt
Priority:       0
Node:           <none>
Labels:         app=containerized-data-importer
                cdi.kubevirt.io=importer
                prometheus.cdi.kubevirt.io=
Annotations:    cdi.kubevirt.io/storage.createdByController: yes
                openshift.io/scc: containerized-data-importer
Status:         Pending
IP:             
IPs:            <none>
Controlled By:  PersistentVolumeClaim/fedora-datavolume-original
Containers:
  importer:
    Image:      kubevirt/cdi-importer:v1.24.0
    Port:       8443/TCP
    Host Port:  0/TCP
    Args:
      -v=1
    Limits:
      cpu:     0
      memory:  0
    Requests:
      cpu:     0
      memory:  0
    Environment:
      IMPORTER_SOURCE:        registry
      IMPORTER_ENDPOINT:      docker://quay.io/kubevirt/common-templates:fedora
      IMPORTER_CONTENTTYPE:   kubevirt
      IMPORTER_IMAGE_SIZE:    15Gi
      OWNER_UID:              186c5b78-0df0-49dd-ba9d-e5d4c98f5f80
      FILESYSTEM_OVERHEAD:    0.055
      INSECURE_TLS:           false
      IMPORTER_DISK_ID:       
      IMPORTER_UUID:          
      IMPORTER_BACKING_FILE:  
      IMPORTER_THUMBPRINT:    
    Mounts:
      /data from cdi-data-vol (rw)
      /scratch from cdi-scratch-vol (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-sdshb (ro)
Conditions:
  Type           Status
  PodScheduled   False 
Volumes:
  cdi-data-vol:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  fedora-datavolume-original
    ReadOnly:   false
  cdi-scratch-vol:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  fedora-datavolume-original-scratch
    ReadOnly:   false
  default-token-sdshb:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-sdshb
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  kubernetes.io/os=linux
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type     Reason            Age        From  Message
  ----     ------            ----       ----  -------
  Warning  FailedScheduling  <unknown>        persistentvolumeclaim "fedora-datavolume-original-scratch" not found
  Warning  FailedScheduling  <unknown>        error while running "VolumeBinding" prebind plugin for pod "importer-fedora-datavolume-original": Failed to bind volumes: provisioning failed for PVC "fedora-datavolume-original"
  Warning  FailedScheduling  <unknown>        error while running "VolumeBinding" prebind plugin for pod "importer-fedora-datavolume-original": Failed to bind volumes: provisioning failed for PVC "fedora-datavolume-original"
```
```
[ksimon@localhost common-templates]$ oc describe dv fedora-datavolume-original
Name:         fedora-datavolume-original
Namespace:    kubevirt
Labels:       <none>
Annotations:  API Version:  cdi.kubevirt.io/v1beta1
Kind:         DataVolume
Metadata:
  Creation Timestamp:  2020-10-06T09:39:11Z
  Generation:          4
  Managed Fields:
    API Version:  cdi.kubevirt.io/v1beta1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .:
          f:kubectl.kubernetes.io/last-applied-configuration:
      f:spec:
        .:
        f:pvc:
          .:
          f:accessModes:
          f:resources:
            .:
            f:requests:
              .:
              f:storage:
        f:source:
          .:
          f:registry:
            .:
            f:url:
    Manager:      oc
    Operation:    Update
    Time:         2020-10-06T09:39:11Z
    API Version:  cdi.kubevirt.io/v1beta1
    Fields Type:  FieldsV1
    fieldsV1:
      f:status:
        .:
        f:conditions:
        f:phase:
        f:progress:
    Manager:         cdi-controller
    Operation:       Update
    Time:            2020-10-06T09:39:27Z
  Resource Version:  40414
  Self Link:         /apis/cdi.kubevirt.io/v1beta1/namespaces/kubevirt/datavolumes/fedora-datavolume-original
  UID:               186c5b78-0df0-49dd-ba9d-e5d4c98f5f80
Spec:
  Pvc:
    Access Modes:
      ReadWriteMany
    Resources:
      Requests:
        Storage:  15Gi
  Source:
    Registry:
      URL:  docker://quay.io/kubevirt/common-templates:fedora
Status:
  Conditions:
    Last Heartbeat Time:   2020-10-06T09:39:27Z
    Last Transition Time:  2020-10-06T09:39:11Z
    Message:               PVC fedora-datavolume-original Pending
    Reason:                Pending
    Status:                False
    Type:                  Bound
    Last Heartbeat Time:   2020-10-06T09:39:11Z
    Last Transition Time:  2020-10-06T09:39:11Z
    Status:                False
    Type:                  Ready
    Last Heartbeat Time:   2020-10-06T09:39:11Z
    Last Transition Time:  2020-10-06T09:39:11Z
    Status:                False
    Type:                  Running
  Phase:                   Pending
  Progress:                N/A
Events:
  Type    Reason   Age                   From                   Message
  ----    ------   ----                  ----                   -------
  Normal  Pending  7m6s                  datavolume-controller  target PVC fedora-datavolume-original Pending and Claim Pending
  Normal  Pending  6m51s (x2 over 7m7s)  datavolume-controller  PVC fedora-datavolume-original Pending
```
```
[ksimon@localhost common-templates]$ oc describe pvc
Name:          fedora-datavolume-original
Namespace:     kubevirt
StorageClass:  managed-premium
Status:        Pending
Volume:        
Labels:        app=containerized-data-importer
Annotations:   cdi.kubevirt.io/storage.condition.bound: true
               cdi.kubevirt.io/storage.condition.bound.message: 
               cdi.kubevirt.io/storage.condition.bound.reason: 
               cdi.kubevirt.io/storage.contentType: 
               cdi.kubevirt.io/storage.import.endpoint: docker://quay.io/kubevirt/common-templates:fedora
               cdi.kubevirt.io/storage.import.importPodName: importer-fedora-datavolume-original
               cdi.kubevirt.io/storage.import.requiresScratch: true
               cdi.kubevirt.io/storage.import.source: registry
               cdi.kubevirt.io/storage.pod.phase: Pending
               cdi.kubevirt.io/storage.pod.restarts: 0
               volume.beta.kubernetes.io/storage-provisioner: kubernetes.io/azure-disk
               volume.kubernetes.io/selected-node: ci-ln-w22h0lb-002ac-7tslj-worker-centralus1-rjhds
Finalizers:    [kubernetes.io/pvc-protection]
Capacity:      
Access Modes:  
VolumeMode:    Filesystem
Mounted By:    importer-fedora-datavolume-original
Events:
  Type     Reason                Age                    From                         Message
  ----     ------                ----                   ----                         -------
  Normal   WaitForFirstConsumer  8m4s (x2 over 8m4s)    persistentvolume-controller  waiting for first consumer to be created before binding
  Warning  ProvisioningFailed    2m37s (x10 over 8m4s)  persistentvolume-controller  Failed to provision volume with StorageClass "managed-premium": invalid AccessModes [ReadWriteMany]: only AccessModes [ReadWriteOnce] are supported with maxShares(0) < 2
  Normal   WaitForPodScheduled   22s (x14 over 8m4s)    persistentvolume-controller  waiting for pod importer-fedora-datavolume-original to be scheduled


Name:          fedora-datavolume-original-scratch
Namespace:     kubevirt
StorageClass:  managed-premium
Status:        Bound
Volume:        pvc-343841d8-fda2-46a5-ba05-14e8901e2e28
Labels:        app=containerized-data-importer
Annotations:   pv.kubernetes.io/bind-completed: yes
               pv.kubernetes.io/bound-by-controller: yes
               volume.beta.kubernetes.io/storage-provisioner: kubernetes.io/azure-disk
               volume.kubernetes.io/selected-node: ci-ln-w22h0lb-002ac-7tslj-worker-centralus1-rjhds
Finalizers:    [kubernetes.io/pvc-protection]
Capacity:      15Gi
Access Modes:  RWO
VolumeMode:    Filesystem
Mounted By:    importer-fedora-datavolume-original
Events:
  Type    Reason                 Age    From                         Message
  ----    ------                 ----   ----                         -------
  Normal  WaitForPodScheduled    8m4s   persistentvolume-controller  waiting for pod importer-fedora-datavolume-original to be scheduled
  Normal  ProvisioningSucceeded  7m53s  persistentvolume-controller  Successfully provisioned volume pvc-343841d8-fda2-46a5-ba05-14e8901e2e28 using kubernetes.io/azure-disk
```
CDI Operator log
[operator.log](https://github.com/kubevirt/containerized-data-importer/files/5332876/operator.log)
CDI Deployment log
[deployment.log](https://github.com/kubevirt/containerized-data-importer/files/5332893/deployment.log)

**Comment by awels** [comment]
Your problem is here: Warning  ProvisioningFailed    2m37s (x10 over 8m4s)  persistentvolume-controller  Failed to provision volume with StorageClass "managed-premium": invalid AccessModes [ReadWriteMany]: only AccessModes [ReadWriteOnce] are supported with maxShares(0) < 2
You are trying to create a RWX DV (and thus PVC) while your provisioner only supports RWO.

Note there is a known bug in the registry import, we are tracking the fix with #1413. Use an older CDI (1.22 for instance) that doesn't have the problem, once you have fixed the request to not ask for a RWX volume.

**Comment by ksimon1** [comment]
Thank for explanation. I am closing this bug.
